test/org/zaproxy/zap/extension/ascanrulesBeta/CsrftokenscanTest.java

```diff
@@ -0,0 +1,174 @@
+package org.zaproxy.zap.extension.ascanrulesBeta;
+
+import org.junit.Test;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.model.OptionsParam;
+import org.parosproxy.paros.network.HtmlParameter;
+import org.parosproxy.paros.network.HttpMalformedHeaderException;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.httpsessions.HttpSessionToken;
+import org.zaproxy.zap.extension.httpsessions.HttpSessionsParam;
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.TreeSet;
+
+public class CsrftokenscanTest extends ActiveScannerTest<Csrftokenscan> {
+
+    @Override
+    protected Csrftokenscan createScanner() {
+        Csrftokenscan scanner = new Csrftokenscan();
+        scanner.setConfig(getConfigWithHTTPSession("asp.net_sessionid", "aspsessionid", "siteserver", "cfid",
```

Comment:
I'd suggest using other configuration, this seems to imply that you need to set one with the HTTP session tokens but the scanners use a different configuration (read from scan policy file) than the main options.

Reply:
Should be ok now

Reply:
Sorry for not being clear, I was suggesting using a new configuration instead (`getConfigWithHTTPSession` would be called in another statement).

Reply:
Let me understand that point : we change `getConfigWithHTTPSession()` in `createScanner()` by `new ZapXmlConfiguration`, so we have to call `getConfigWithHTTPSession()` for each test method using authentication ?

Reply:
More or less, `getConfigWithHTTPSession()` can still be called in this method just not set into the scanner. (The method `getConfigWithHTTPSession` can also be renamed to `setupHttpSessionsParam`?)
```Java
@Override
protected Csrftokenscan createScanner() {
    Csrftokenscan scanner = new Csrftokenscan();
    scanner.setConfig(new ZapXmlConfiguration());
    setUpHttpSessionsParam();
    return scanner;
}
```

Reply:
thanks, I ,didin't see the point of duplicating the call, now I understand :)